### PR TITLE
Armor Class adjustments

### PR DESCRIPTION
**Pull request by feliniddotexe**
## About The Pull Request

This PR adjusts some armor-related stuff.

- Adds armor class flags for leg armors.

- Leather/Heavy Leather armor is considered light class, chainmail armor is medium, plate armor is heavy. Bikinis/skirts remain a bit lighter at the cost of less armor coverage than actual chausses/shirts.

- In keeping with the above, changes chainmail shirts/haubergeons to medium armor. Light classes will have to use gambesons under their leathers!

- Adjusts loadouts for a handful of classes that spawn with chainmail stuff without medium/heavy armor proficiency. (Namely squires, desert riders, and the prince/consort.)

- Some smaller tweaks to armor values. Leather armors and gambesons are now slightly better at negating blunt damage than chainmail, but fare worse against cuts/stabs.

## Why It's Good For The Game

Light armor classes are meant to be faster and harder to hit but less durable than classes with heavy armor. As is, the only armor that actually uses armor flags is chest armor, though - so light armor classes still wind up being pretty tanky if they kit themselves out appropriately. It's possible to wear all of the same gear as a heavy armor class and still benefit from dodge expert, save for the chestpiece - which is pretty silly. You can swap out the chestpiece for studded leather and a haubergeon and that's still around 500 durability worth of chest armor!

Now that we have a few extra leather armor options thanks to #850 I've gone ahead and added armor class flags for leggings and made some other smaller tweaks to armor values as a whole. This makes the progression from light to medium to heavy armor pretty straightforward. Leathers are considered light armor, chainmail is considered medium, and plate is considered heavy - simple as that.

I considered potentially adding armor class flags for things like bracers and helmets as well, but unfortunately we really don't have enough light armor options for those armor slots - so I'm holding off for now.

### CODE REVIEW

**Comment by Ruediger4** [comment]
I LOVE the idea but I can't help but squint at two things

1) Chainmaille shirts (not HAUBERGEONS, just iron shirts) are still somewhat light and I *think* could maybe stay as light. Haubergeon as medium? Fine, though I dont see why someone wouldn't just get a hauberk then, but I think iron maille could stay as light.

2) Some numbers are like. Massive. I think boosting some of the armor up to 80 blunt prot is a lot. 70 may be the sweet spot?

File: code/modules/jobs/job_types/roguetown/mercenaries/classes/desertrider.dm [hunk 1]
```diff
@@ -70,8 +71,9 @@
 			H.change_stat("intelligence", 1)
 			H.change_stat("speed", 3)
 			backl = /obj/item/rogueweapon/sword/long/rider
-			shirt = /obj/item/clothing/suit/roguetown/shirt/undershirt/black
-			armor = /obj/item/clothing/suit/roguetown/armor/chainmail
+			shirt = /obj/item/clothing/suit/roguetown/armor/gambeson
+			armor = /obj/item/clothing/suit/roguetown/armor/leather/heavy/coat
+			pants = /obj/item/clothing/under/roguetown/trou/leather
```
Comment (kittysmooch) [inline code/modules/jobs/job_types/roguetown/mercenaries/classes/desertrider.dm:76]: this variant is totally unarmored, they have armored leather pants that you can give them now

File: code/modules/jobs/job_types/roguetown/youngfolk/squire.dm [hunk 2]
```diff
@@ -131,6 +131,7 @@
 /datum/outfit/job/roguetown/squire/skirmisher/pre_equip(mob/living/carbon/human/H)
 	beltr = /obj/item/quiver/arrows
 	armor = /obj/item/clothing/suit/roguetown/armor/leather/studded
+	pants = /obj/item/clothing/under/roguetown/trou/leather
```
Comment (kittysmooch) [inline code/modules/jobs/job_types/roguetown/youngfolk/squire.dm:134]: same here

**Review by kittysmooch — COMMENTED** [review]
we HATE leather pantaloons

**Comment by LittleEvil9** [comment]
I would say not to change the leather armors integrity, its already somewhat easy to cut it to pieces due to having low resistances compared to other armor classes.

**Comment by LittleEvil9** [comment]
> Gambesons & leather armors in general got a slight buff to bludgeoning resistance and are otherwise mostly unchanged.
> 
> Studded leather armor is still 300 durability, the same as before. The only integrity values I changed were the new heavy leather armors. I gave them just a _bit_ less durability than studded leather because they're relatively easily craftable and don't require a smith to make, unlike studded leather. They have the exact same protection stats as studded leather otherwise - just 50 less durability, which is realistically only like two extra hits.

They're actually a major annoyance to make, 3 leather, 1 fat and 3 fibers for both chest armor and pants, 4 leather and 2 fat if you want to make the leather coat armor. The price is higher than any studded leather you would purchase in the smith, needing 10+ int or skincraft skill to make. If you *want* to hunt for these components you have to spend time making lots of sweet bait and hunting for ages, because no aggressive ambush mobs drop fats when butchered.

We wanted more leather armor options and we finally have them, heavy leather armor was made to be an equivalent to studded leather due to the crippling lack of armor choices and to be a decent armor for self sufficient hunters, do not nerf their durability stats. Let people be creative.

> 
> The only other 'nerfed' item would be the heavy leather pants - which I changed because they literally had the exact same stats as iron chainmail leggings but with 50 extra durability over iron chainmail leggings for some reason - they were objectively better than chainmail and that was kind of silly.

Yeah thats fine, imo they should both have the *same* durability but with leather pants having slightly less defence, since you know, leather and all. 

I could argue about keeping their stats the same with less durability since leg meta is the most annoying part for most to deal with, but the only thing I dislike rn is the durability changes explained above.

**Comment by kittysmooch** [comment]
i swear i gave the heavy leather pants lower stats than chausses but idk maybe i forgor

**Comment by kittysmooch** [comment]
> Increased heavy leather armor's durability a bit to be closer to studded leather. Shifted from 250 to 280. Realistically this means that studded leather can maybe take _one_ extra hit over heavy leather given it's made with iron but it's pretty much on par with studded leather.

also keep in mind that even though you can technically scavenge heavy leather's components in practice it takes a long time for bait to roll an animal with fat, and if you buy the materials from the stockpile the heavy leather recipe actually costs a bit more.

**Comment by kittysmooch** [comment]
if you removed leg protection then you should probably remove the extra craft cost ig

**Comment by Lutowski** [comment]
EPIC!!!!!!!!!!!!!!!!!!!!!!

**Pull request merged**